src/lib/util.js

```diff
@@ -6,11 +6,22 @@ export function defined(obj) {
 }
 
 
+/** A simpler Object.assign
+ *  @private
+ */
+export function assign(obj, props) {
```

Comment:
This PR looks good.  The only reason why I kept `Object.assign` originally instead of doing a simply polyfill and using `h` instead of spread operators in JSX  was because I was concerned that a year from now when someone comes back and touches this that we would forget that we can't use those.  What are your thoughts on what this does to maintainability?